grafanalib/core.py

```diff
@@ -1577,3 +1593,144 @@ def to_json_data(self):
+
+    :param allValue: If All values should be shown or a Calculation
+    :param cacheTimeout: metric query result cache ttl
+    :param calc: Calculation to perform on metrics
+    :param dataSource: Grafana datasource name
+    :param decimals: override automatic decimal precision for legend/tooltips
+    :param description: optional panel description
+    :param editable: defines if panel is editable via web interfaces
+    :param format: defines value units
+    :param height: defines panel height
+    :param hideTimeOverride: hides time overrides
+    :param id: panel id
+    :param interval: defines time interval between metric queries
+    :param labels: oprion to show gauge level labels
+    :param limit: limit of number of values to show when not Calculating
+    :param links: additional web links
+    :param max: maximum value of the gauge
+    :param maxDataPoints: maximum metric query results,
+        that will be used for rendering
+    :param min: minimum value of the gauge
+    :param minSpan: minimum span number
+    :param rangeMaps: the list of value to text mappings
+    :param span: defines the number of spans that will be used for panel
+    :param targets: list of metric requests for chosen datasource
+    :param thresholdLabel: label for gauge. Template Variables:
+        "$__series_namei" "$__field_name" "$__cell_{N} / $__calc"
+    :param thresholdMarkers: option to show marker of level on gauge
+    :param thresholds: single stat thresholds
+    :param timeFrom: time range that Override relative time
+    :param title: panel title
+    :param transparent: defines if panel should be transparent
+    :param valueMaps: the list of value to text mappings
+    """
+
+    title = attr.ib()
+    targets = attr.ib()
+    allValues = attr.ib(default=False, validator=instance_of(bool))
+    cacheTimeout = attr.ib(default=None)
+    calc = attr.ib(default=GAUGE_CALC_MEAN)
+    dataSource = attr.ib(default=None)
+    decimals = attr.ib(default=None)
+    description = attr.ib(default=None)
+    editable = attr.ib(default=True, validator=instance_of(bool))
+    format = attr.ib(default="none")
+    height = attr.ib(default=None)
+    hideTimeOverride = attr.ib(default=False, validator=instance_of(bool))
+    id = attr.ib(default=None)
+    interval = attr.ib(default=None)
+    label = attr.ib(default=None)
+    limit = attr.ib(default=None)
+    links = attr.ib(default=attr.Factory(list))
+    max = attr.ib(default=100)
+    maxDataPoints = attr.ib(default=100)
+    min = attr.ib(default=0)
+    minSpan = attr.ib(default=None)
+    rangeMaps = attr.ib(default=attr.Factory(list))
+    repeat = attr.ib(default=None)
+    span = attr.ib(default=6)
+    thresholdLabels = attr.ib(default=False, validator=instance_of(bool))
+    thresholdMarkers = attr.ib(default=True, validator=instance_of(bool))
+    thresholds = attr.ib(
+        default=attr.Factory(
+            lambda: [
+                Threshold("green", 0),
```

Comment:
Threshold `value` is currently a required positional argument, so this default breaks. Either give `value` a default value, or provide a `value` here.